api/server/server.go

```diff
@@ -187,18 +190,18 @@ func (s *Server) startGears(ctx context.Context) {
 	// By default it serves on :8080 unless a
 	// PORT environment variable was defined.
 	svr.AddFunc(func(ctx context.Context) {
-		go s.Router.Run()
+		listen := fmt.Sprintf(":%d", viper.GetInt(EnvPort))
+		listener, err := net.Listen("tcp", listen)
+		if err != nil {
+			logrus.WithError(err).Fatalln("Failed to serve functions API.")
+		}
+		logrus.Infof("Serving Functions API on address `%s`", listen)
+		go runner.StartWorkers(ctx, s.Runner, s.tasks)
+		go runner.RunAsyncRunner(ctx, s.apiURL, s.tasks, s.Runner)
+		go http.Serve(listener, s.Router)
```

Comment:
This is all wrong. srv.AddFunc should be managing individual services. Here it is managing all of them together. 👎